content/examples/flatcar-container-linux/_index.md

```diff
@@ -0,0 +1,221 @@
+
+Example snippet of hardware entry with `gateway` defined:
+
+```json
+...
+   "ip_addresses" : [
+      {
+         "address" : "192.168.50.10",
+         "address_family" : 4,
+         "gateway" : "192.168.50.2",
+         "netmask" : "255.255.255.0"
+      }
+   ]
+...
+```
+
+### Use local mirror
+
+If your workers do not have internet access or if you want to speed up the provisioning process,
+you can set up a local Flatcar mirror, from where the Flatcar images will be pulled by workers.
+The images will be served by nginx running as part of Tinkerbell stack.
+The mirror can be created and updated using the [official script](https://github.com/kinvolk/flatcar-release-mirror).
+
+#### Creating mirror
+
+To prepare the directories for the mirror and download the mirroring script, run the following command:
+
+```sh
+mkdir /var/tinkerbell/nginx/misc/flatcar
+cd /var/tinkerbell/nginx/misc/flatcar
+wget https://github.com/kinvolk/flatcar-release-mirror/raw/master/flatcar-release-mirror.sh && chmod +x flatcar-release-mirror.sh
+```
+
+Then, run the mirror script:
+
+```shell
+./flatcar-release-mirror.sh --only-files 'flatcar_production_image\|version' --above-version 2344 --logfile /dev/stderr --channels stable
+```
+
+It will download all Flatcar images above version 2344 only from stable channel. This currently requires downloading around 1.1 GB of data.
+
+If you want to download only the latest version, you can check the Flatcar releases [here](https://www.flatcar-linux.org/releases/).
+
+Please note, that downloading only specific channel/versions may limit the usage of the `--channel` and `--version`
+parameters in `flatcar-install`.
+
+#### Using local mirror in the template
+
+To use local mirror in the template, add the following arguments to the `flatcar-install` script in the `flatcar-install` task described in the next step:
+
+```yaml
+- -b
+- http://192.168.50.3/misc/flatcar/stable/amd64-usr
+```
+
+Where `192.168.50.3` is the Tinkerbell nginx IP address, which can be obtained from
+`envrc` file generated by `setup.sh` script, as `TINKERBELL_NGINX_IP` variable, for example
+by running the following command (if you previously run `source envrc`):
+
+```sh
+echo $TINKERBELL_NGINX_IP
+```
+
+## Preparing template
+
+Here is an example template, which can be used as a base for the next steps.
+
+```yaml
+version: "0.1"
+name: flatcar-install
+global_timeout: 1800
+tasks:
+  - name: "flatcar-install"
+    worker: "{{.machine1}}"
+    volumes:
+      - /dev:/dev
+      - /statedir:/statedir
+    actions:
+      - name: "dump-ignition"
+        image: flatcar-install
+        command:
+          - sh
+          - -c
+          - echo 'eyJpZ25pdGlvbiI6eyJjb25maWciOnt9LCJzZWN1cml0eSI6eyJ0bHMiOnt9fSwidGltZW91dHMiOnt9LCJ2ZXJzaW9uIjoiMi4yLjAifSwibmV0d29ya2QiOnt9LCJwYXNzd2QiOnt9LCJzdG9yYWdlIjp7fSwic3lzdGVtZCI6e319' | base64 -d > /statedir/ignition.json
```

Comment:
isn't there a way to push this somewhere else? its really odd to have instance specific data here. shouldn't this be done when we are using `workflow create` instead (using e.g. `{{.ignition}}` here)?

Reply:
I know this is not the ideal place to put it, but from my understanding of Tinkerbell architecture, this is the best place to put it. Or is is now maybe allowed to add arbitrary data while creating the workflows?

Reply:
You can add data when creating the workflow; like the {{.machine1}} that is in line 136.

Reply:
>You can add data when creating the workflow; like the {{.machine1}} that is in line 136.

I just tested it and indeed it seems to work, thanks! Previously, when there was still `target` around, this was not possible. I'll update the guide.

>Maybe we can also push this to the metadata service as some kind of secret?

I don't know how this could be done. Could you elaborate your idea please?